### PR TITLE
Remove redundant fallback for view_as_complex

### PR DESCRIPTION
**Pull request by int3**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103261

This enables lowering to work for it

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @aakhundov

Differential Revision: [D46585029](https://our.internmc.facebook.com/intern/diff/D46585029)

### CODE REVIEW

**Review by desertfire — APPROVED** [review]
Thanks for fixing it.

File: torch/_inductor/lowering.py [hunk 1]
```diff
@@ -1491,7 +1491,7 @@ def apply_constraint(arg, fx_arg):
 make_fallback(aten.topk)
 make_fallback(aten.upsample_bicubic2d_backward, require_contiguous)
 
-make_fallback(aten.view_as_complex.default, require_contiguous)
+make_fallback(aten.view_as_complex, require_contiguous)
```
Comment (eellison) [inline torch/_inductor/lowering.py:1494]: We shouldn't need this change to make it work, they are equivalent

**Review by eellison — APPROVED** [review]
Looks good but we should make the overloaded registration work too

File: torch/_inductor/lowering.py (hunk 1 above)
Comment (int3) [reply torch/_inductor/lowering.py:1494]: Without that change we get

```
LoweringException: AssertionError: view_as_complex.default is not supported with cpp wrapper
```

Comment (desertfire) [reply torch/_inductor/lowering.py:1494]: It's being fixed in https://github.com/pytorch/pytorch/issues/102752. Let's not duplicate work.

**Comment by int3** [comment]
@int3 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D46585029).

**Comment by int3** [comment]
@int3 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D46585029).

**Pull request closed without merging**